DFE-Digital/schools-experience
# Se 1786/placement request validations

### JIRA Ticket Number
SE-1786

### Context
Fixes a couple of issues around schools updating subject specific dates with placement requests.

### Changes proposed in this pull request
Defaults the selected subject in confirm booking to be the subject specific date's subject if applicable otherwise the first teaching choice (placment_request#requested_subject)
Fix an issue around schools changing a date from flex to fix by making the validation on bookings_placement_date_id contextual.
Fix an issue around schools changing a date to be subject specific by making the validation on bookings_placement_dates_subject_id contextual.
Fix an issue around schools removing a subject from a subject specific date by storing the subject_id on placement_request rather than storing the bookings_placement_dates_subject_id

### Guidance to review
Manual testing:
Create a placement request for a subject specific date, expect to be able to remove the subject for the subject specific date.
